vm-virtio/src/lib.rs

```diff
@@ -132,20 +198,35 @@ impl Translatable for u64 {
 }
 
 /// Helper for cloning a Queue since QueueState doesn't derive Clone
-pub fn clone_queue(queue: &Queue) -> Queue {
+pub fn clone_queue(queue: &Queue, access_platform: Option<Arc<dyn AccessPlatform>>) -> Queue {
     let mut q = Queue::new(queue.max_size()).unwrap();
 
     q.set_next_avail(queue.next_avail());
     q.set_next_used(queue.next_used());
     q.set_event_idx(queue.event_idx_enabled());
     q.set_size(queue.size());
     q.set_ready(queue.ready());
-    q.try_set_desc_table_address(GuestAddress(queue.desc_table()))
-        .unwrap();
-    q.try_set_avail_ring_address(GuestAddress(queue.avail_ring()))
-        .unwrap();
-    q.try_set_used_ring_address(GuestAddress(queue.used_ring()))
-        .unwrap();
+
+    let mut desc_addr = GuestAddress(queue.desc_table());
```

Comment:
The need for adding this confuses me - since `virtio-iommu` would have required such virtio table translations. I can see that in the existing code this translation happens in `VirtioPciCommonConfig::write_common_config_word`:

```rust
                if let Some(access_platform) = &self.access_platform {
                    if ready {
                        let desc_table = access_platform.translate_gva(q.desc_table(), 0).unwrap();
                        let avail_ring = access_platform.translate_gva(q.avail_ring(), 0).unwrap();
                        let used_ring = access_platform.translate_gva(q.used_ring(), 0).unwrap();
                        q.set_desc_table_address(
                            Some((desc_table & 0xffff_ffff) as u32),
                            Some((desc_table >> 32) as u32),
                        );
                        q.set_avail_ring_address(
                            Some((avail_ring & 0xffff_ffff) as u32),
                            Some((avail_ring >> 32) as u32),
                        );
                        q.set_used_ring_address(
                            Some((used_ring & 0xffff_ffff) as u32),
                            Some((used_ring >> 32) as u32),
                        );
                    }
```

Since this is done when the virtio PCI device is marked as ready I think we need to understand why it's also needs to be done here - and if instead we should be doing it one (better) place (rather than now two.)

Reply:
I think I got your point, I have refactored code to consolidate code in one place.